src/main/java/org/terasology/climateConditions/igloo/IglooProvider.java

```diff
@@ -0,0 +1,102 @@
+
+package org.terasology.climateConditions.igloo;
+
+import org.terasology.core.world.CoreBiome;
+import org.terasology.core.world.generator.facets.BiomeFacet;
+import org.terasology.math.TeraMath;
+import org.terasology.math.geom.Rect2i;
+import org.terasology.math.geom.Vector2i;
+import org.terasology.utilities.procedural.Noise;
+import org.terasology.utilities.procedural.WhiteNoise;
+import org.terasology.world.generation.Border3D;
+import org.terasology.world.generation.Facet;
+import org.terasology.world.generation.FacetBorder;
+import org.terasology.world.generation.FacetProviderPlugin;
+import org.terasology.world.generation.GeneratingRegion;
+import org.terasology.world.generation.Produces;
+import org.terasology.world.generation.Requires;
+import org.terasology.world.generation.facets.SeaLevelFacet;
+import org.terasology.world.generation.facets.SurfaceHeightFacet;
+import org.terasology.world.generation.facets.base.BaseFieldFacet2D;
+import org.terasology.world.generator.plugin.RegisterPlugin;
+
+@RegisterPlugin
+@Requires({
+        @Facet(value = SurfaceHeightFacet.class, border = @FacetBorder(sides = Igloo.SIZE * 2)),
+        @Facet(value = BiomeFacet.class, border = @FacetBorder(sides = Igloo.SIZE * 2)),
+        @Facet(value = SeaLevelFacet.class, border = @FacetBorder(sides = Igloo.SIZE * 2))
+})
+@Produces(IglooFacet.class)
+public class IglooProvider implements FacetProviderPlugin {
+    private Noise noise;
+
+    @Override
+    public void setSeed(long seed) {
+        noise = new WhiteNoise(seed + 3);
+    }
+
+    /**
+     * Places the Igloo Structure in the Snow Biome with a very low probability of spawn.
+     */
+    @Override
+    public void process(GeneratingRegion region) {
+
+        Border3D border = region.getBorderForFacet(IglooFacet.class).extendBy(0, Igloo.SIZE * 2,
+                Igloo.SIZE * 2);
+
+        IglooFacet facet = new IglooFacet(region.getRegion(), border);
+        SurfaceHeightFacet surfaceHeightFacet = region.getRegionFacet(SurfaceHeightFacet.class);
+        SeaLevelFacet seaLevelFacet = region.getRegionFacet(SeaLevelFacet.class);
+        BiomeFacet biomeFacet = region.getRegionFacet(BiomeFacet.class);
+
+        Rect2i worldRegion = surfaceHeightFacet.getWorldRegion();
+
+        for (int wz = worldRegion.minY(); wz <= worldRegion.maxY(); wz++) {
+            for (int wx = worldRegion.minX(); wx <= worldRegion.maxX(); wx++) {
+                int surfaceHeight = TeraMath.floorToInt(surfaceHeightFacet.getWorld(wx, wz));
+                int seaLevel = seaLevelFacet.getSeaLevel();
+                // check if height is within this region
+                if (surfaceHeight >= facet.getWorldRegion().minY() &&
+                        surfaceHeight <= facet.getWorldRegion().maxY()) {
+                    // Sea Level + 96 is the height at which snow blocks are placed in all biomes.
+                    if (noise.noise(wx, wz) > 0.9999f && (surfaceHeight >= seaLevel + 96f ||
```

Comment:
this is some implicit knowledge from the implementation of our current default world rasterizer. I'd like to add at least a `//TODO: ...` comment here to derive the height from other properties.

Reply:
I'm a little confused.., do you mean the seaLevel + 96 height form some other class right?
If that's the case, I think the change should first be made in the solid rasterizer: https://github.com/Terasology/CoreWorlds/blob/4419b057d731dbbd35e523c11c227f2dd9599843/src/main/java/org/terasology/core/world/generator/rasterizers/SolidRasterizer.java#L116
and define a constant like SNOW_BIOME_THRESHOLD = seaLevel + 96 and use that here like 
>                     } else if (depth == 0 && height >= SNOW_BIOME_THRESHOLD) {

and then use that in here(in igloo provider)

Reply:
If I understand it right joining in late here then yeah if `96` is the same number here and in CoreWorlds IMHO it would make sense to make a constant over there then also use that constant here. 

At the same time I concur with the desire to _eventually_ get away from such crude ways to designate a given height as "Okay magically now it is snow due to height!" disregarding other climate details - but yeah that might be hard to really do much about here at the moment other than maybe leaving a TODO or two for future considerations :-)